platform/lib/build/pageTransformer.js

```diff
@@ -116,15 +119,108 @@ class PageMinifier {
           canonicalPage.contents = Buffer.from(optimizedHtml);
           ampPage.contents = Buffer.from(html);
 
-          this.push(canonicalPage);
-          this.push(ampPage);
+          let filteredPages = [];
+          if (isFilterableRoute(canonicalPage.path)) {
+            filteredPages = scope._filterPages(canonicalPage, ampPage);
+          }
 
+          for (const page of [canonicalPage, ampPage, ...filteredPages]) {
+            this.push(page);
+          }
+
+          scope._log.success(`Transformed ${canonicalPage.path}`);
           callback();
         }))
         .pipe(gulp.dest('./'));
   }
 
 
+  /**
+   * Verifies a given page should be filtered
+   *
+   * @param  {Vinyl}
+   * @return {undefined}
+   */
+  _isManuallyFiltered(page) {
+    let path = page.path.replace('.amp.html', '.html');
+
+    // Skip pages that have been manually filtered and therefore have a path like
+    // - guides-and-tutorials/index.websites.html
+    // - guides-and-tutorials/index.email.amp.html
+    // But still the matching format should be filtered, therefore pass it back
+    const format = path.match(/\.(websites|stories|ads|email)(\.amp)?\.html/);
+    if (format) {
+      return format[1];
+    }
+
+    // Do not filter pages that have a manually filtered equivalent as they
+    // are also somewhere in the stream and shouldn't be overwritten
+    path = path.replace('.html', `.${format}.html`);
+    if (fs.existsSync(path)) {
+      return true;
+    }
+
+    return false;
+  }
+
+  /**
+   * Filters the given pages by the formats defined in data-available-formats
+   *
+   * @param  {Array} ...pages
+   * @return {Array}
+   */
+  _filterPages(...pages) {
+    const filteredPages = [];
+    for (const page of pages) {
+      const html = page.contents.toString();
+      const manualFormat = this._isManuallyFiltered(page);
+      if (manualFormat) {
+        const filteredHtml = this.filterHtml(html, manualFormat, true);
+        page.contents = Buffer.from(filteredHtml);
+        continue;
+      }
+
+      for (const format of FORMATS) {
+        const filteredHtml = this.filterHtml(html, format);
+        if (filteredHtml) {
+          const filteredPage = page.clone();
+          filteredPage.contents = Buffer.from(filteredHtml);
+
+          // As websites is the default those files can be overwritten and
+          // don't need an extra name
+          if (format !== 'websites') {
+            filteredPage.stem = filteredPage.stem.replace('.', `.${format}.`);
+          }
+
+          filteredPages.push(filteredPage);
+        }
+      }
+    }
+
+    return filteredPages;
+  }
+
+  /**
+   * @param  {String} html
+   * @return {String}
+   */
+  filterHtml(html, format, force) {
+    const dom = cheerio.load(html);
+    if (filterPage(format, dom, force)) {
```

Comment:
nit: using  an early exit avoids indentation

Reply:
it's also weird that only one code path returns a value

Reply:
`filterPage` now returns `true|false`.